mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioSelector.java

```diff
@@ -247,6 +247,11 @@ public void run() {
                 int workCount = 0;
                 long beforeSelect = System.nanoTime();
                 int selected = select(selector, quickSelect);
+                // The SelectorUtil.EPOLL_BUG_WORKAROUND condition was removed in Netty 3.10.5 and instead
+                // added to the if (selectReturnsImmediately == 1024) condition later on. This seems inefficient
+                // for the (common) case where the workaround is not enabled since in that case there's no point
+                // in looping through the selector keys, so we are keeping the condition here. There's no risk
+                // of a busy loop (https://github.com/netty/netty/issues/2426) when the workaround is not enabled.
                 if (SelectorUtil.EPOLL_BUG_WORKAROUND && selected == 0 && !wakenupFromLoop && !wakenUp.get()) {
```

Comment:
I don't mind removing this whole if condition. I don't think we will enable EPOLL_BUG_WORKAROUND. My understanding is that this for very old JDKs